product/optee-stm32mp1/fw/config_scmi_voltage_domains.c

```diff
@@ -0,0 +1,262 @@
+/*
+ * Arm SCP/MCP Software
+ * Copyright (c) 2022, Linaro Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <fwk_element.h>
+#include <fwk_id.h>
+#include <fwk_module.h>
+#include <fwk_module_idx.h>
+
+#include <mod_voltage_domain.h>
+#include <mod_scmi_voltage_domain.h>
+#include <mod_stm32_pmic_regu.h>
+#include <mod_stm32_pwr_regu.h>
+#include <scmi_agents.h>
+
+#include <assert.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <compiler.h>
+#include <drivers/stm32mp1_pwr.h>
+#include <dt-bindings/regulator/st,stm32mp15-regulator.h>
+#include <util.h>
+
+/*
+ * stm32_pwr_cfg - Configuration data for PWR regulators exposed thru SCMI
+ *
+ * These configation data are referenced in the fwk config data of
+ * fwk modules VOLTAGE_DOMAIN and STM32_PMIC_REGU.
+ */
+enum stm32_pwr_regu {
+    STM32_PWR_REG11,
+    STM32_PWR_REG18,
+    STM32_PWR_USB33,
+};
+
+static const struct mod_stm32_pwr_regu_dev_config stm32_pwr_cfg[] = {
+    [STM32_PWR_REG11] = { .pwr_id = PWR_REG11, .regu_name = "reg11" },
+    [STM32_PWR_REG18] = { .pwr_id = PWR_REG18, .regu_name = "reg18" },
+    [STM32_PWR_USB33] = { .pwr_id = PWR_USB33, .regu_name = "usb33" },
+};
+
+/*
+ * stm32_pmic_cfg - Configuration data for PMIC regulators exposed thru SCMI
+ *
+ * These configation data are referenced in the fwk config data of
```

Comment:
Typo 

I realise "data" is a plural but normally I would opt for "this configuration data is referenced", even though I know it is a plural really :)